cli/dcos-spark/submit_builder.go

```diff
@@ -483,59 +492,30 @@ func buildSubmitJson(cmd *SparkCommand, marathonConfig map[string]interface{}) (
 		args.properties["spark.app.name"] = args.mainClass
 	}
 
-	// driver image
-	var imageSource string
+	// driver image: use provided value
 	_, contains := args.properties["spark.mesos.executor.docker.image"]
-	if contains {
-		imageSource = "Spark config: spark.mesos.executor.docker.image"
-	} else {
-		if cmd.submitDockerImage == "" {
-			dispatcher_image, err := getStringFromTree(marathonConfig, []string{"app", "container", "docker", "image"})
-			if err != nil {
-				return "", err
-			}
-			args.properties["spark.mesos.executor.docker.image"] = dispatcher_image
-			imageSource = "dispatcher: container.docker.image"
-		} else {
-			args.properties["spark.mesos.executor.docker.image"] = cmd.submitDockerImage
-			imageSource = "flag: --docker-image"
-		}
```

Comment:
I think removing this section will change the existing behavior: if a special Docker image was provided when installing the Dispatcher, that same image will also be used for Drivers and/or Executors (see the logic below). Unless we make some changes in the Dispatcher.